src/pages/dataElements/edit/formatFormValues.ts

```diff
@@ -0,0 +1,51 @@
+import get from 'lodash/fp/get'
+import { JsonPatchOperation } from '../../../types'
+import { DataElement } from '../../../types/generated'
+import { FormValues } from '../form/types'
+
+type DataElementKey = keyof DataElement
+
+interface FormatFormValuesArgs {
+    dataElement: DataElement
+    dirtyFields: { [name: string]: boolean }
+    values: FormValues
+}
+
+const sanitizeDirtyValyeKeys = (keys: DataElementKey[]) => {
+    const attributeValuesDirty = keys.find((key) =>
+        key.startsWith('attributeValues')
+    )
+
+    if (!attributeValuesDirty) {
+        return keys
+    }
+
+    return [
+        ...keys.filter((key) => !key.startsWith('attributeValues')),
+        'attributeValues' as DataElementKey,
+    ]
+}
+
+export function formatFormValues({
```

Comment:
Sorry about nickpicking on names. But I think this could be more explicit to convey that it formats form values to json-patch.

Reply:
Changed it to `createJsonPatchOperations`